test/rules-tealium-fullstory.spec.ts

```diff
@@ -0,0 +1,90 @@
+import 'mocha';
+
+import DataHandler from '../src/handler';
+import { rules } from '../examples/rules/tealium-fullstory.json';
+
+import { tealiumRetail } from './mocks/tealium';
+import {
+  expectEqual, expectRule, expectFS, setupGlobals, ExpectObserver, expectNoCalls, expectGlobal, expectUndefined,
+  expectMatch,
+} from './utils/mocha';
+
+describe('Tealium to FullStory rules', () => {
+  beforeEach(() => setupGlobals([
+    ['utag', { data: tealiumRetail }], // NOTE the expando becomes utag.data
+    ['_dlo_rules', rules],
+  ]));
+
+  afterEach(() => {
+    ExpectObserver.getInstance().cleanup();
+  });
+
+  it('should read tealium_event', () => {
+    expectGlobal('utag').data.tealium_event = 'product_view';
+
+    ExpectObserver.getInstance().create({
+      rules: [expectRule('fs-tealium-event')],
+    });
+
+    const [id, payload] = expectFS('event');
+    expectEqual(id, 'product_view');
+    expectMatch(payload, tealiumRetail, 'product_id');
+    expectEqual(payload.order_total, 54.47);
+    expectEqual(payload.product_discount_amount, 2.98); // NOTE list reduced to single value
+    expectUndefined(payload, 'customer_first_name', 'customer_last_name', 'customer_email');
+  });
+
+  it('should not monitor properties not included in the source', (done) => {
+    expectGlobal('utag').data.tealium_event = 'product_view';
+
+    ExpectObserver.getInstance().create({
+      // ensure this rule never reads on load
+      rules: [{ ...expectRule('fs-tealium-event'), readOnLoad: false }],
+    });
+
+    // NOTE this is a valid property to monitor
+    expectGlobal('utag').data.product_id = ['PROD789'];
+
+    // check the assignment
+    setTimeout(() => {
```

Comment:
why run these assertions in a `setTimeout`?

Reply:
The `DataHandler` debounces the incoming events so that multiple assignments to an object's properties don't fire N `FS.events`.  For example, say the `cart` was sequentially assigned `total = 12.99` and `tax = 1.99`.  It waits for a timeout period when no new assignments have occurred and then sends the final `cart` object with the two updated properties.

But this is a good example of where this has been a copy/paste job across a few tests that should be codified in a util function.  I created a card to track.